client.go

```diff
@@ -0,0 +1,280 @@
+package globalsign
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+)
+
+// Client globasign client.
+type Client struct {
+	sync.RWMutex
+	// httpClient used to communicate with the API.
+	httpClient *http.Client
+
+	// BaseURL base URL of API.
+	BaseURL *url.URL
+
+	// userAgent user agent for client header request.
+	userAgent string
+
+	// authToken token for authorization.
+	authToken *string
+
+	// authTokenTs authentication token timestamp.
+	authTokenTs time.Time
+
+	// DSSService Digital Signing Service (DSS).
+	DSSService DSSService
+
+	// vault store identity information.
+	vault *IdentityVault
+
+	// options globalsign client options.
+	options *ClientOptions
+}
+
+// ClientOptions options for the globalsign client.
+type ClientOptions struct {
+	// BaseURL base url of API.
+	BaseURL *url.URL
+
+	// ApiKey API key credentials.
+	ApiKey string
+
+	// ApiSecret API secret credentials.
+	ApiSecret string
+
+	// CertFilePath path file to mTLS cert file.
+	CertFilePath string
+
+	// KeyFilePath path file to mTLS cert private key.
+	KeyFilePath string
+}
+
+// NewClient initiates client with `API Key`, `API Secret`, `Certificate file path`, `Private Key file path` and returns the globalsign client.
```

Comment:
normally only put `variable` names inside ticks.  Can just skip the ticks I think for this

Reply:
Updated at 6fa006e290519f3ae08c512c53671a7b8ce4e71f